src/ui/Misc/TextEllipsisTooltip.ts

```diff
@@ -0,0 +1,47 @@
+import 'styling/_TextEllipsisTooltip';
+import PopperJs from 'popper.js';
+import { $$ } from '../../utils/Dom';
+
+export class TextEllipsisTooltip {
+  public element: HTMLElement;
+  private arrow: HTMLElement;
+
+  constructor(private textReference: HTMLElement, private caption: string, private root: HTMLElement) {
+    this.element = $$('div', { className: 'coveo-text-ellipsis-tooltip coveo-hidden' }, this.caption).el;
+    this.arrow = $$('div', { className: 'coveo-text-ellipsis-tooltip-arrow' }).el;
+
+    this.element.appendChild(this.arrow);
+
+    this.buildPopper();
+  }
+
+  private buildPopper() {
+    const popperReference = new PopperJs(this.textReference, this.element, {
+      placement: 'top',
+      modifiers: {
+        preventOverflow: {
+          boundariesElement: this.root,
+          padding: 10
+        },
+        arrow: {
+          element: this.arrow
+        },
+        // X,Y offset of the tooltip relative to the icon
+        offset: {
+          offset: '0,8'
+        }
+      }
+    });
+
+    $$(this.textReference).on('mouseenter', () => {
+      if (this.textReference.offsetWidth < this.textReference.scrollWidth) {
```

Comment:
I have a feeling this logic will get pretty complicated overtime, depending on different browsers and different CSS rules that clients might inject in their page.

Having this work 100% of the time in all scenarios for all browsers will be challenging.
What if there's different padding/border added in a given page with customization ?

Those CSS rules affect this calculation, meaning there will be corner case where it does not work properly.